### ISSUE TITLE
broken links in ePub

### ISSUE THREAD

**Issue by peterstadler** [TEIC/Stylesheets#375]
## what's wrong

The recent merge of the PR #336 introduced an issue with internal chapters links. The ePub file `Guidelines.epub` is actually a zip archive with the following structure:

```
- META_INF
- OPS
  | - AB.html
  | - AI.html
  | - BIB.html
  | - …
- mimetype
```
See, e.g. the file `AB.html` and compare
the old, correct output as of ed6ac726
```html
<a class="link_ref" href="ST.html#STGAre" title="Rendition Indicators">the rendition indicators</a>
```

with the current dev branch
```html
<a href="#STGAre">the rendition indicators</a>
```

Strangely enough, only the epub transformation seems to be broken while the html transformation looks ok.

## how to reproduce

checkout old version of the Stylesheets by issuing 
```bash
git checkout ed6ac72
```
and then build the Guidelines.epub (against these Stylesheets) within the TEI repo by entering
```bash
git checkout dev
git pull
cd P5
make epub
```
and compare the output of the files within the archive with the [last successful build](https://jenkins.tei-c.org/job/TEIP5-dev/lastSuccessfulBuild/artifact/P5/release/doc/tei-p5-doc/en/) from Jenkins

**Comment by jamescummings** [comment]
Thanks for the issue. So somewhere there _used_ to be processing of links for HTML that did something differently and now the new processing doesn't build the bit with the filename?

**Comment by peterstadler** [comment]
@jamescummings No, the standard HTML processing is not affected (because it does not include `odds/teiodds.xsl` which was changed in #336)! The epub processing, on the other hand, does include  `odds/teiodds.xsl`, see https://github.com/TEIC/Stylesheets/blob/4bdb73d806014992a2e5ea56f58834d3c61bcb72/profiles/tei/epub3/to.xsl#L16

**Comment by jamescummings** [comment]
Ah, I see. And partly that is because one of the things being produced by the EPUB conversion is a copy of the Guidelines?

**Comment by peterstadler** [comment]
Since the original intention of #336 was to only alter the RelaxNG output, Stylesheets group decided on 2019-05-20 to better move the newly introduced templates from `teiodds.xsl` to `odd2relaxng.xsl`.

**Comment by peterstadler** [comment]
I think that's been fixed by 2a01be4 but we should double check when the build is ok again.

**Comment by peterstadler** [comment]
build is ok, ePub looks good :)
